src/data/move.ts

```diff
@@ -1880,17 +1882,37 @@ export class StealEatBerryAttr extends EatBerryAttr {
     if (heldBerries.length) { // if the target has berries, pick a random berry and steal it
       this.chosenBerry = heldBerries[user.randSeedInt(heldBerries.length)];
 
-      if (this.chosenBerry.stackCount === 1) { // remove modifier if its the last berry
+      if (this.chosenBerry === undefined) { // if no berry has been provided, pick a random berry from their inventory
```

Comment:
```suggestion
      if (!this.chosenBerry) { // if no berry has been provided, pick a random berry from their inventory
```
don't check for `undefined` UNLESS you really want to exclude `null`